sqlalchemy_bigquery/base.py

```diff
@@ -655,25 +656,79 @@ def get_column_specification(self, column, **kwargs):
-                bq_opts.get("description", table.comment)
+        partition_by_or_cluster_by_or_options = []
+
+        if "partitioning" in bq_opts:
+            partition_expr = bq_opts.get("partitioning")
+
+            assert isinstance(
+                partition_expr, str
+            ), f"bigquery_partitioning dialect option accepts only {str}, provided {repr(partition_expr)}"
+
+            partition_by_or_cluster_by_or_options.append(
+                f"PARTITION BY {partition_expr}"
             )
-            opts.append(f"description={description}")
 
-        if "friendly_name" in bq_opts:
-            opts.append(
-                "friendly_name={}".format(
-                    process_string_literal(bq_opts["friendly_name"])
-                )
+        if "clustering_fields" in bq_opts:
+            clustering_fields = bq_opts.get("clustering_fields")
+
+            assert isinstance(
+                clustering_fields, list
+            ), f"bigquery_clustering_fields dialect option accepts only {list}, provided {repr(clustering_fields)}"
+
+            for field in clustering_fields:
+                if field not in table.c:
+                    raise NoSuchColumnError(field)
+
+            partition_by_or_cluster_by_or_options.append(
+                f"CLUSTER BY {', '.join(clustering_fields)}"
             )
 
-        if opts:
-            return "\nOPTIONS({})".format(", ".join(opts))
+        options = {}
 
-        return ""
+        if ("description" in bq_opts) or table.comment:
+            description = bq_opts.get("description", table.comment)
+            options["description"] = description
+
+        table_option_list = {
+            "friendly_name": str,
+            "expiration_timestamp": datetime.datetime,
+            "partition_expiration_days": int,
+            "require_partition_filter": bool,
+            "default_rounding_mode": str,
+        }
+
+        for table_option in table_option_list:
+            if table_option in bq_opts:
+                options[table_option] = bq_opts.get(table_option)
+
+        if options:
+
+            def process_option_value(option, value):
+                if option in table_option_list:
+                    assert isinstance(
+                        value, table_option_list[option]
+                    ), f"bigquery_{option} dialect option accepts only {table_option_list[option]}, provided {repr(value)}"
+                else:
+                    return process_string_literal(value)
+
+                if table_option_list[option] == str:
+                    return process_string_literal(value)
+                if table_option_list[option] == int:
+                    return int(value)
+                if table_option_list[option] == bool:
+                    return "true" if value else "false"
+                if table_option_list[option] == datetime.datetime:
+                    return BQTimestamp.process_timestamp_literal(value)
```

Comment:
Maybe this could be modified to a dict with a function for each type?

```
option_casting = {
    # Mapping from option type to its casting method
    str: lambda value: process_string_literal(value),
    int: lambda value: int(value),
    bool: lambda value: "true" if value else "false",
    datetime.datetime: lambda value: BQTimestamp.process_timestamp_literal(value),
}

if (option_type := table_option_list.get(option) is None:
    return process_string_literal(value)
assert isinstance(
     value, option_type
), f"bigquery_{option} dialect option accepts only {table_option_list[option]}, provided {repr(value)}"

# TODO: Can also add assert for option_type in option_casting
if (option_cast := option_casting.get(option_type)) is not None:
    return option_cast(value)
```

Reply:
Sure we could do this, but when the set of choices if finite I would rather go with if / return statement.
Using a dict makes more sense when we expect new entries in the future. This way, the additionnal complexity is balanced by the limited effort required to add new capabilities.

Reply:
OK, so strong feelings here about this way or another.

One thing, though, maybe just add some code that handles the fallback if no type was matched.

Reply:
@snapiri Yes, you're right. The coverage is not ok with this, I'm working on it.

Reply:
@snapiri
I had considered the possibility of recommending a dict with functions as a substitute for a series of if statements, but I wanted to minimize change in this PR. It is large enough as it is. 

There is also a potential decrease in readability depending on how the code modified.

Reply:
@chalmerlowe @snapiri 

I finally decided to split the function into 2 distinct functions that convey better their intention.

https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f756cb148bee9c81c91362eed3059807e24c43dd/sqlalchemy_bigquery/base.py#L745-L793

Tell me what you think.

I can still change the if statements to a dict if you both think this is the recommended way, it doesn't take much effort.